### PR TITLE
CSS modules: Callout, ContextualMenu, Tooltip, TeachingBubble, Dialog

### PR DESCRIPTION
**Pull request by dzearing**
This update patches the scss to use css modules for these components:

Callout
ContextualMenu
Tooltip
TeachingBubble
Dialog

Note: this PR is for the `module-css-updates` branch. We'll merge all changes there, publish a beta build, test some projects with that, then merge into master and publish after we've done some internal testing.

To make the PR easier to read, you can disable whitespace on the diff by adding the ?w=1 parameter to the PR url:

https://github.com/OfficeDev/office-ui-fabric-react/pull/1049/files?w=1

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Dialog/Dialog.scss [hunk 1]
```diff
@@ -17,209 +17,197 @@ $Dialog-default-max-width: 340px;
...
 
-  //= Modifier: Multiline button dialog
-  //
-  .ms-Dialog.ms-Dialog--multiline {
-    .ms-Dialog-title {
-      font-size: $ms-font-size-xxl;
-    }
+//= Modifier: Multiline button dialog
+//
+.root.isMultiline {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Dialog/Dialog.scss]: `.rootIsMultiline`, unless we want to wait and do the 'flattening' in a separate PR.

File: packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.scss [hunk 2]
```diff
@@ -139,7 +139,7 @@
     @include focus-border();
     min-width: 0;
   }
-  .ms-MessageBar-actionsOneline .ms-MessageBar-dismissal {
+  .ms-MessageBar-dismissalOneline .ms-MessageBar-dismissal {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.scss]: This change is also showing in another PR.

File: packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.tsx [hunk 3]
```diff
@@ -4,12 +4,14 @@ import * as React from 'react';
...
   BaseComponent,
   autobind,
+  css,
   divProperties,
   getNativeProps
 } from '../../Utilities';
 import { ITooltipHostProps } from './TooltipHost.Props';
 import { Tooltip } from './Tooltip';
 import { TooltipDelay } from './Tooltip.Props';
+import styles from './Tooltip.scss';
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.tsx]: If TooltipHost is a separate component, it should have its own stylesheet right?

File: packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.tsx [hunk 4]
```diff
@@ -35,7 +37,7 @@ export class TooltipHost extends BaseComponent<ITooltipHostProps, any> {
     let { isTooltipVisible } = this.state;
     return (
       <div
-        className='ms-TooltipHost'
+        className={ css('ms-TooltipHost', styles.host) }
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.tsx]: `styles.root` once this has its own stylesheet.

File: packages/office-ui-fabric-react/src/components/TeachingBubble/TeachingBubbleContent.tsx [hunk 5]
```diff
@@ -57,9 +57,11 @@ export class TeachingBubbleContent extends BaseComponent<ITeachingBubbleProps, I
       headerContent = (
         <div className={ css(
           'ms-TeachingBubble-header',
-          hasCondensedHeadline ? 'ms-TeachingBubble-header--small' : 'ms-TeachingBubble-header--large'
+          hasCondensedHeadline ?
+            'ms-TeachingBubble-header--small ' + styles.isSmall :
+            'ms-TeachingBubble-header--large ' + styles.isLarge
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/TeachingBubble/TeachingBubbleContent.tsx]: headerIsSmall etc

File: packages/office-ui-fabric-react/src/components/Dialog/Dialog.scss (hunk 1 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dialog/Dialog.scss]: not going to change any specificity if i don't have ot.

File: packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.tsx (hunk 3 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.tsx]: There are many places where this applies and i'm avoiding unnecessary changes.

File: packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.tsx (hunk 4 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.tsx]: not gonna split this out for 1 rule for a subcomponent, let's get these in with minimal changes.
